### PR TITLE
Update pytest_plugin with fixtures to test auth in core and extensions

### PR DESCRIPTION
**Pull request by akshaychitneni**
Fixes https://github.com/jupyter-server/jupyter_server/issues/773

### CODE REVIEW

**Review by blink1073 — DISMISSED** [review]
Nice, thanks @akshaychitneni!

File: jupyter_server/pytest_plugin.py [hunk 1]
```diff
@@ -494,3 +498,123 @@ async def _():
...
+        try:
+            r = await fetch(url, **fetch_kwargs, allow_nonstandard_methods=True)
+            code = r.code
+        except HTTPClientError as err:
+            code = err.code
+        else:
+            if fetch is jp_ws_fetch:
+                r.close()
+
+        print(code, url, fetch_kwargs)
```
Comment (Zsailer) [inline jupyter_server/pytest_plugin.py:522]: Should this be removed? 😉

**Review by Zsailer — DISMISSED** [review]
Thanks @akshaychitneni! Just a minor comment inline. Otherwise this looks good!

**Comment by akshaychitneni** [comment]
> Thanks @akshaychitneni! Just a minor comment inline. Otherwise this looks good!

Thanks @Zsailer. Updated.

**Pull request merged**